pkg/distributor/distributor.go

```diff
@@ -2061,8 +2062,13 @@ func (d *Distributor) MetricsForLabelMatchers(ctx context.Context, from, through
 		}
 	}
 
+	queryLimiter := mimir_limiter.QueryLimiterFromContextWithFallback(ctx)
+
 	result := make([]labels.Labels, 0, len(metrics))
 	for _, m := range metrics {
+		if err := queryLimiter.AddSeries(mimirpb.FromLabelsToLabelAdapters(m)); err != nil {
```

Comment:
Assuming that this conversion isn't too expensive?

Reply:
It's type conversion only: https://github.com/grafana/mimir/blob/c60f33203d52d9b667a18a98592f1a0c8741f1a1/pkg/mimirpb/compat_slice.go#L86